python/lsst/obs/base/ingest.py

```diff
@@ -554,6 +587,7 @@ def run(self, files, *, pool: Optional[Pool] = None, processes: int = 1, run: Op
             try:
                 self.butler.registry.syncDimensionData("exposure", exposure.record)
             except Exception as e:
+                self._on_ingest_failure(exposure, e)
```

Comment:
Will exposure here allow me to get the path to the file that failed to ingest?   My use case is that on ingest failure, I move the file to a special directory so it can be looked at later.

Reply:
As you can see by the warning issued below, this is a per-exposure failure and not a per file failure. I report the obsid and instrument name so the set of 189 files can be identified, but it's no longer a per-file error.

Reply:
I took a look at the whole file.  I can use exposure.files to get at that information though, right?

Reply:
Ah yes, it's a `RawExposureData`.  Serves me right for not reading higher. Per-file metadata failures are explicitly a file failure higher up.

Reply:
I'm going to have to change all this to be ButlerURI based imminently.

Reply:
If you've already got a branch with `ButlerURI` conversions, I'm happy to rebase on that and make the adjustments.  If not, I assume you'd just like me to turn this around quickly?

And yes, you can get the files from the object the callback is given here, but it'll be all of the files you gave it for each exposure at once, because we commit them all in one transaction.

Reply:
I haven't got that branch. I have got #357 which is quite big but it's possible that that is orthogonal to your changes. I won't be able to merge #357 for a while because the review is still pending so you can go ahead. I'll take a look in the next half hour once I get immutable ButlerURI finished up.